### ISSUE TITLE
Incorrect ReturnType of `DurableOrchestrationClient.getStatus`

### ISSUE THREAD

**Issue by christianemmert** [Azure/azure-functions-durable-js#486]
**Describe the bug**
`DurableOrchestrationClient` interface states that `getStatus()` returns `Promise<DurableOrchestrationStatus>`. This is correct for instances that exist and weren't purged. But for purged instances `getStatus()` returns an empty string. This is highly unexpected and should be fixed.

**Investigative information**

- Durable Functions extension version: 4.0.5030
- durable-functions npm module version: 1.5.4
- Language (JavaScript/TypeScript) and version: 4.8.4 (Typescript)
- Node.js version: 16.15.1

**To Reproduce**

```javascript
const client = getClient(context);
const instanceId = await client.startNew("MY_ORCHESTRATOR");

await client.getStatus(instanceId) // -> returns DurableOrchestrationStatus object

await client.purgeInstanceHistory(instanceId)
await client.getStatus(instanceId) // -> returns ""
```

**Expected behavior**
Interface of `DurableOrchestrationClient.getStatus()` represents the actual ReturnType of the method.

**Comment by davidmrdavid** [comment]
FYI @hossam-nasr

**Comment by christianemmert** [comment]
The problem is in https://github.com/Azure/azure-functions-durable-js/blob/2857fc579ba61b91673e5a4b259bf9ce2386bd1c/src/durableorchestrationclient.ts#L229. When no instance is found the client will return a status code of 404 as such `response.data` won't have the shape of `DurableOrchestrationClient` but rather an empty string.

**Comment by hossam-nasr** [comment]
Fixed by #512
